### DISCUSSION TITLE
How to make vcpkg use (s)ccache from cmake preset

### DISCUSSION THREAD

**Discussion by Arndt-MBE** [microsoft/vcpkg#37800] [Q&A] [answered]
I use `vcpkg` in manifest mode in my project, which is configured by `cmake` presets.
Integration of `vcpkg` via `CMAKE_TOOLCHAIN_FILE` works fine, and the compiler cache (`sccache` in my case) is integrated via `CMAKE_CXX_COMPILER_LAUNCHER` in the actual build process.
But I have not yet found out how to make `vcpkg` use `sccache` as the `CMAKE_CXX_COMPILER_LAUNCHER` in its `cmake` builds.
I read that `vcpkg` unsets all `CMAKE_*` environment variables, so I tried to `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` with the content `set(CMAKE_CXX_COMPILER_LAUNCHER "sccache")`, but that has no effect.

Do you know a good way to make `vcpkg` accept a `CMAKE_CXX_COMPILER_LAUNCHER`?

**Comment by Neumann-A** [comment, marked as answer]
Variables from your CMake project are not forwarded to vcpkg. You need to adjust the toolchain vcpkg internally uses.

**Reply by Arndt-MBE** [reply]
Isn't that what `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` does, like through an official backdoor?

**Reply by Neumann-A** [reply]
No. Only if it is directly set in the triplet.

That variable is meant to load another toolchain alongside vcpkg.cmake for your project.

**Comment by Arndt-MBE** [comment]
#30191 is about a very similar question, although not specific to presets.
